src/closure.js

```diff
@@ -31,28 +36,50 @@ const limitFunctionCallCount = (cb, n) => {
   };
   // Should return a function that invokes `cb`.
   // The returned function should only allow `cb` to be invoked `n` times.
+  let stuff=0;
+  return (...args) => {
+     if (stuff < n) {
+       stuff++;
+       return cb(...args);
+     }
+     return null;
 };
 
 const cacheFunction = (cb) => {
+        // Should return a funciton that invokes `cb`.
+        // A cache (object) should be kept in closure scope.
+        // The cache should keep track of all arguments have been used to invoke this function.
+        // If the returned function is invoked with arguments that it has already seen
+        // then it should return the cached result and not invoke `cb` again.
+        // `cb` should only ever be invoked once for a given set of arguments.
   const cache = {};
-  return (input) => {
-    if (Object.prototype.hasOwnProperty.call(cache, input)) return cache[input];
-    cache[input] = cb(input);
-    return cache[input];
-  };
-  // Should return a funciton that invokes `cb`.
-  // A cache (object) should be kept in closure scope.
-  // The cache should keep track of all arguments have been used to invoke this function.
-  // If the returned function is invoked with arguments that it has already seen
-  // then it should return the cached result and not invoke `cb` again.
-  // `cb` should only ever be invoked once for a given set of arguments.
-};
+ 
+
+        let cacheObject = {
+            //put keys here
+        };
+        return (...args) => {
+            let key = args.toString() + cb.toString();
+            if (!cacheObject[key]) {
+                let value = cb(args)
+                cacheObject[key] = value;
+              console.log(cacheObject)
+                return value;
```

Comment:
If you have a `return` inside of an `if` statement then you actually don't need an `else`. If the `if` ever passes the function will end and the rest of the function will never be reached.